Add Redis requirements to Reference Architecture page (#1566)
wandb/docs: content/en/guides/hosting/hosting-options/self-managed/ref-arch.md

- **Monitoring.** The database should be monitored for load. If CPU usage is sustained at > 40% of the system for more than 5 minutes it is likely a good indication the server is resource starved.
- **Availability.** Depending on your availability and durability requirements you might want to configure a hot standby on a separate machine that streams all updates in realtime from the primary server and can be used to failover to in the event that the primary server crashes or become corrupted.

### Redis

> W&B depends on a single-node Redis 7.x deployment to handle job queuing and data caching used by W&B's components. For convenience during testing and development of proofs of concept, W&B Self-Managed includes a local Redis deployment that is not appropriate for production deployments.

Reviewer: ```suggestion
W&B depends on a single-node Redis 7.x deployment used by W&B's components for job queuing and data caching. For convenience during testing and development of proofs of concept, W&B Self-Managed includes a local Redis deployment that is not appropriate for production deployments.
```

- **Monitoring.** The database should be monitored for load. If CPU usage is sustained at > 40% of the system for more than 5 minutes it is likely a good indication the server is resource starved.
- **Availability.** Depending on your availability and durability requirements you might want to configure a hot standby on a separate machine that streams all updates in realtime from the primary server and can be used to failover to in the event that the primary server crashes or become corrupted.

### Redis

> W&B requires Redis as a job queuing and data caching layer. This is to support the various microservices that run as part of the application.

Reviewer: ```suggestion
W&B depends on a single-node Redis 7.x deployment to handle job queuing and data caching used by W&B's components. For convenience during testing and development of proofs of concept, W&B Self-Managed includes a local Redis deployment that is not appropriate for production deployments.
```

> Currently W&B only supports connecting to a single node Redis deployment running on version 7.x.

Reviewer: ```suggestion
```

> This deployment can be deployed in one of the following:

Reviewer: ```suggestion
W&B can connect to a Redis instance in the following environments:
```

- [AWS Elasticache](https://aws.amazon.com/pm/elasticache/)
- [Google Cloud Memory Store](https://cloud.google.com/memorystore?hl=en)
- [Azure Cache for Redis](https://azure.microsoft.com/en-us/products/cache)
- Redis deployment hosted in your cloud or on-premise infrastructure

> 

Reviewer: ```suggestion
```
